text/0006-shipyard-next-generation.md

```diff
@@ -0,0 +1,317 @@
+      "strategy": "blue_green",
+      "traffic": "100",
+    }
+  }
+}
+```
+
+*Implementation details:* 
+* Keptn-services can add any information to the reserved space. For example:
+  ```
+  {
+    "type": "sh.keptn.event.deployment.triggered",
+    ...
+    "data": {
+      "deployment": {
+        "strategy": "blue_green",
+        "traffic": "100",
+        "deploymentURI":"https://my-service.domain.com/" // Property added by Keptn-service
+      }
+    }
+  }
+  ```
+
+* The event selectors for Keptn-services, as proposed in [First version of a Keptn uniform](https://github.com/keptn/enhancement-proposals/pull/7), work based on the information provided in this reserved space. For example:
+  ```yaml
+    - name: deployment-service
+      image: keptn/helm-service:0.6.0
+      events:
+      - deployment.triggered:
+        selector:
+          matchExpressions:
+          - {key: strategy, operator: In, values: [direct, blue_green]}
+  ```
+
+* The shipyard controller merges the reserved spaces of a `[task-xyz].triggered`, `[task-xyz].started` and `[task-xyz].finished` event. If, for example, the `deployment.triggered` event contains `strategy:blue_green` and the `deployment.finished` event contains `deploymentURI:xyz` in the reserved space, the shipyard controller merges the the reserved space, before sending the next `[test].triggered`: 
+  ```
+  {
+    "type": "sh.keptn.event.task-abc.triggered",
+    ...
+    "data": {
+      "deployment": {
+        "strategy": "blue_green",
+        "traffic": "100",
+        "deploymentURI":"https://my-service.domain.com/" // Property added by Keptn-service
+      }
+      "test": {
+
+      }
+    }
+  }
+  ```
+
+
+
+## Breaking changes
+
+This KEP breaks the implementation of Keptn 0.6.0: 
+* Keptn-services (aka. batteries) including contributed services need to be changed to react on a `[task].triggered`, and send a `[task].started` → `[task].finished`.
+* Keptn CloudEvents must be changed.
+* A shipyard migration must take place as outlined below.
+
+### Upgrade path from Shipyard 0.1.2  to 0.2.0
+
+The shipyard specification version 0.1.2 as used by Keptn 0.5.0 and 0.6.0 (last supported Keptn versions) has to be migrated to the new specification. Thus, the following spec: 
+
+```yaml
+stages:
+- name: "dev"
+  deployment_strategy: "direct"
+  test_strategy: "functional"
+- name: "staging"
+  deployment_strategy: "blue_green_service"
+  test_strategy: "performance"
+- name: "production"
+  deployment_strategy: "blue_green_service"
+  remediation_strategy: "automated"
+```
+
+... migrates to:
+
+```yaml
+---
+version: 0.2.0
+kind: Shipyard
+metadata:
+  name: shipyard
+spec:
+  stages:
+  - name: "dev"
+    sequences:
+    - name: artifact-delivery
+      tasks:
+      - name: deployment
+        properties:
+          strategy: direct
+      - name: test
+        properties:
+          kind: functional
+      - name: evaluation 
+      - name: release
+
+  - name: "hardening"
```

Comment:
@johannes-b the second stage is called "staging" in the old yaml

Reply:
good catch

Reply:
thx, fixed it.